commit f0de3cec40ee0f365751e11a477e47dd5568c34e
Author: Johan Hedberg <johan.hedberg@intel.com>
Date:   Fri Mar 11 09:56:33 2016 +0200

    Bluetooth: Fix potential buffer overflow with Add Advertising
    
    commit 6a0e78072c2ae7b20b14e0249d8108441ea928d2 upstream.
    
    The Add Advertising command handler does the appropriate checks for
    the AD and Scan Response data, however fails to take into account the
    general length of the mgmt command itself, which could lead to
    potential buffer overflows. This patch adds the necessary check that
    the mgmt command length is consistent with the given ad and scan_rsp
    lengths.
    
    Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>
    Signed-off-by: Marcel Holtmann <marcel@holtmann.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/bluetooth/mgmt.c b/net/bluetooth/mgmt.c
index 7f22119..b1b0a1c 100644
--- a/net/bluetooth/mgmt.c
+++ b/net/bluetooth/mgmt.c
@@ -7155,6 +7155,10 @@ static int add_advertising(struct sock *sk, struct hci_dev *hdev,
 		return mgmt_cmd_status(sk, hdev->id, MGMT_OP_ADD_ADVERTISING,
 				       status);
 
+	if (data_len != sizeof(*cp) + cp->adv_data_len + cp->scan_rsp_len)
+		return mgmt_cmd_status(sk, hdev->id, MGMT_OP_ADD_ADVERTISING,
+				       MGMT_STATUS_INVALID_PARAMS);
+
 	flags = __le32_to_cpu(cp->flags);
 	timeout = __le16_to_cpu(cp->timeout);
 	duration = __le16_to_cpu(cp->duration);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1369_f0de3cec40ee0f365751e11a477e47dd5568c34e.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1369_f0de3cec40ee0f365751e11a477e47dd5568c34e.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
